File: LondonFhirService.Manage/Models/Foundations/Patients/StructuredRecordRequest.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

namespace LondonFhirService.Manage.Models.Foundations.Patients
{
    /// <summary>
    /// What the structured record page posts. The four credential fields are optional: each one
    /// falls back to its PatientConfiguration counterpart when the operator leaves it blank, so a
    /// caller can exercise the configured consumer or supply another one without a redeploy.
    ///
    /// No member carries a validation attribute. Every rule this request is held to lives in
    /// PatientService.Validations, so a rejection is one the foundation states rather than one
    /// model binding invented before the service was reached.
    /// </summary>
    public class StructuredRecordRequest
    {
        public string ClientId { get; set; }
        public string ClientSecret { get; set; }
        public string Scope { get; set; }
        public string GrantType { get; set; }
        public string NhsNumber { get; set; }
        public string DateOfBirth { get; set; }
        public bool DemographicsOnly { get; set; }
    }
}

File: LondonFhirService.Manage.Client/src/models/foundations/patients/StructuredRecordRequest.ts
// What PatientsController accepts on the wire. The four credential fields are optional to the
// server - a blank one falls back to the value the Manage host is configured with - which is why
// none of them are required by the form's own validation either.
export type StructuredRecordRequest = {
    clientId: string;
    clientSecret: string;
    scope: string;
    grantType: string;
    nhsNumber: string;
    dateOfBirth: string;
    demographicsOnly: boolean;
};
